force-app/main/default/lwc/barcodeScannerExample/__tests__/barcodeScannerExample.test.js

```diff
@@ -0,0 +1,90 @@
+import { createElement } from 'lwc';
+import { getNavigateCalledWith } from 'lightning/navigation';
+import BarcodeScannerExample from 'c/barcodeScannerExample';
+import { setBarcodeScannerAvailable } from 'lightning/mobileCapabilities';
+
+describe('c-barcode-scanner-example', () => {
+    afterEach(() => {
+        // The jsdom instance is shared across test cases in a single file so reset the DOM
+        while (document.body.firstChild) {
+            document.body.removeChild(document.body.firstChild);
+        }
+
+        // Prevent data saved on mocks from leaking between tests
+        jest.clearAllMocks();
+    });
+
+    // Helper function to wait until the microtask queue is empty.
+    // Used when having to wait for asynchronous DOM updates.
+    async function flushPromises() {
+        return Promise.resolve();
+    }
+
+    it('directs the user to the mobile app when Barcode Scanner is unavailable', async () => {
+        // Create initial BarcodeScannerExample element and attach to virtual DOM
+        const elementBarcodeScannerExample = createElement(
+            'c-barcode-scanner-example',
+            {
+                is: BarcodeScannerExample
+            }
+        );
+        document.body.appendChild(elementBarcodeScannerExample);
+
+        // Wait for element to mount
+        await flushPromises();
+
+        // Mount `Scan QR Code` button and trigger scan of property record ID
+        const elementScannerDirections =
+            elementBarcodeScannerExample.shadowRoot.querySelector(
+                '#scanner-directions'
```

Comment:
[You must not rely on IDs selectors in LWC. ](https://developer.salesforce.com/docs/component-library/documentation/en/lwc/create_components_dom_work)
Use a class selector instead.